### PR TITLE
Save user language in order

### LINKED ISSUE #265: Save language the order was placed in
Multilanguage sites will need a way to determine the language to send emails in. It's better to follow whatever language the user chose instead of trying to figure it out based on the country.

### PR DESCRIPTION
**Pull request by maciekjablonski**
Fixes #265

### CODE REVIEW

File: saleor/checkout/core.py [hunk 1]
```diff
@@ -235,6 +244,9 @@ def create_order(self):
             'tracking_client_id': self.tracking_code,
             'total': self.get_total()}
 
+        if self.language_code:
```
Comment (patrys) [inline saleor/checkout/core.py]: When is this going to be false?

**Comment by patrys** [comment]
Why do we need to store the language in checkout instead of saving it directly when creating an order?

**Comment by maciekjablonski** [comment]
@patrys i removed language from checkout

**Comment by maciekjablonski** [comment]
@patrys re-review please

File: tests/test_checkout_integration.py [hunk 2]
```diff
@@ -306,3 +306,40 @@ def test_remove_voucher(client, request_cart_with_item, shipping_method, voucher
...
+
+
+def test_language_is_saved_in_order(authorized_client, billing_address, customer_user,  # pylint: disable=R0914
+                                    request_cart_with_item, settings, shipping_method):
+    """
+    authorized user change own email after checkout - if is not changed in order
+    """
+    # Prepare some data
+    user_language = 'fr'
+    assert user_language != settings.LANGUAGE_CODE
```
Comment (patrys) [inline tests/test_checkout_integration.py]: What is this testing?

File: tests/test_checkout_integration.py [hunk 3]
```diff
@@ -306,3 +306,40 @@ def test_remove_voucher(client, request_cart_with_item, shipping_method, voucher
...
+    customer_user.addresses.add(billing_address)
+    request_cart_with_item.user = customer_user
+    request_cart_with_item.save()
+
+    # Enter checkout
+    # Checkout index redirects directly to shipping address step
+    shipping_address = authorized_client.get(reverse('checkout:index'),
+                            follow=True, HTTP_ACCEPT_LANGUAGE=user_language)
+
+    # Enter shipping address data
```
Comment (patrys) [inline tests/test_checkout_integration.py]: You can make the order non-shippable and skip half of the code in this test.

File: tests/test_checkout_integration.py [hunk 4]
```diff
@@ -306,3 +306,40 @@ def test_remove_voucher(client, request_cart_with_item, shipping_method, voucher
...
+    user_language = 'fr'
+    assert user_language != settings.LANGUAGE_CODE
+    customer_user.addresses.add(billing_address)
+    request_cart_with_item.user = customer_user
+    request_cart_with_item.save()
+
+    # Enter checkout
+    # Checkout index redirects directly to shipping address step
+    shipping_address = authorized_client.get(reverse('checkout:index'),
+                            follow=True, HTTP_ACCEPT_LANGUAGE=user_language)
```
Comment (patrys) [inline tests/test_checkout_integration.py]: PEP8?

File: saleor/order/models.py [hunk 5]
```diff
@@ -59,6 +59,7 @@ class Order(models.Model, ItemSet, index.Indexed):
     user = models.ForeignKey(
         settings.AUTH_USER_MODEL, blank=True, null=True, related_name='orders',
         verbose_name=pgettext_lazy('Order field', 'user'))
+    language_code = models.CharField(max_length=5, default=settings.LANGUAGE_CODE)
```
Comment (patrys) [inline saleor/order/models.py]: `zh-hans` is longer than 5 characters.

**Comment by maciekjablonski** [comment]
Changes done

File: saleor/checkout/core.py [hunk 6]
```diff
@@ -197,8 +198,8 @@ def is_shipping_same_as_billing(self):
     def _add_to_user_address_book(self, address, is_billing=False,
                                   is_shipping=False):
         if self.user.is_authenticated():
-            store_user_address(self.user, address, shipping=is_shipping,
-                               billing=is_billing)
+            store_user_address(self.user, address,
+                shipping=is_shipping, billing=is_billing)
```
Comment (krzysztofwolski) [inline saleor/checkout/core.py]: Indent

**Comment by maciekjablonski** [comment]
Language code length reason: https://tools.ietf.org/html/rfc5646#section-4.4.1

**Pull request merged**